### PR TITLE
Update etcd to be a modern version

### PR DESCRIPTION
**Pull request by ajroetker**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,8 @@
 language: go
-env:
-  - GO111MODULE=on
 script: go test -race -cpu 1,2,4 -v -timeout 5m ./...
 sudo: false
 go:
-  - 1.13.x
+  - 1.16.x
```
Comment (mattayes) [inline .travis.yml:5]: We should follow Go's guidelines and support the last two major versions:

```suggestion
  - 1.15.x
  - 1.16.x
```

Or we should make the repo private 😉

File: go.mod [hunk 2]
```diff
@@ -3,31 +3,12 @@ module github.com/lytics/metafora
...
-	golang.org/x/time v0.0.0-20190921001708-c4c64cad1fd0 // indirect
-	google.golang.org/genproto v0.0.0-20190916214212-f660b8655731 // indirect
-	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
+	github.com/araddon/gou v0.0.0-20190110011759-c797efecbb61
+	github.com/kr/pretty v0.2.1 // indirect
+	github.com/kr/text v0.2.0 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	github.com/stretchr/testify v1.7.0
+	go.etcd.io/etcd/client/v3 v3.5.0
+	go.etcd.io/etcd/server/v3 v3.5.0 // indirect
```
Comment (mattayes) [inline go.mod:12]: [Q] Do we have to upgrade etcd to use this or is it backwards-compatible?

File: metcdv3/balancer.go [hunk 3]
```diff
@@ -7,7 +7,7 @@ import (
 
 	"github.com/lytics/metafora"
 
-	etcdv3 "go.etcd.io/etcd/clientv3"
+	etcdv3 "go.etcd.io/etcd/client/v3"
```
Comment (mattayes) [inline metcdv3/balancer.go:10]: [Q] How would we refer to this package if we weren't using the alias? `client`? That might be cleaner?

**Review by jonasagx — APPROVED** [review]

**Review by erinpentecost — APPROVED** [review]

File: metcdv3/balancer.go (hunk 3 above)
Comment (ajroetker) [reply metcdv3/balancer.go:10]: it turns into clientv3

Comment (ajroetker) [reply metcdv3/balancer.go:10]: etcdv3 is easier to read imo since we don't use the server

**Review by mattayes — APPROVED** [review]
:shipit: (if tests are passing)

**Pull request merged**